### PR TITLE
LPS-186788 Remove multiple configuration from LiferayOauthDataProvider

### PR DESCRIPTION
**Pull request by julschong**
resubmitted from #4948 - changed RuntimeException to OAuthServiceException

### CODE REVIEW

**Comment by Kvale1733** [comment]
ci:test:oauth2

**Comment by Kvale1733** [comment]
ci:test:oauth2

**Comment by catarineaguiar** [comment]
no regressions found

File: modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/liferay/LiferayOAuthDataProvider.java [hunk 1]
```diff
@@ -1233,56 +1240,71 @@ private Client _populateClient(
...
+					if (_log.isDebugEnabled()) {
+						_log.debug(
+							"Unknown or disabled grant type " +
+								allowedGrantType);
+					}
 				}
 			}
 		}
+		catch (ConfigurationException configurationException) {
+			throw new RuntimeException(configurationException);
```
Comment (tinatian) [inline modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/liferay/LiferayOAuthDataProvider.java:1306]: @julschong Why this one is still using RuntimeException?

**Pull request closed without merging**